src/data-model-providers/codegen/ClusterIntegration.cpp

```diff
@@ -0,0 +1,140 @@
+/*
+ *    Copyright (c) 2025 Project CHIP Authors
+ *    All rights reserved.
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+#include <data-model-providers/codegen/ClusterIntegration.h>
+
+#include <app/util/attribute-storage-null-handling.h>
+#include <app/util/attribute-storage.h>
+#include <app/util/attribute-table.h>
+#include <app/util/endpoint-config-api.h>
+#include <data-model-providers/codegen/CodegenDataModelProvider.h>
+#include <limits>
+
+namespace chip::app {
+
+namespace {
+
+bool findEndpointWithLog(EndpointId endpointId, ClusterId clusterId, uint16_t fixedClusterServerEndpointCount,
+                         uint16_t maxEndpointCount, uint16_t & zeroBasedArrayIndex)
+{
+    zeroBasedArrayIndex = emberAfGetClusterServerEndpointIndex(endpointId, clusterId, fixedClusterServerEndpointCount);
+
+    if (zeroBasedArrayIndex >= maxEndpointCount)
+    {
+        ChipLogError(AppServer,
+                     "Could not find a valid endpoint index for endpoint %u/" ChipLogFormatMEI " (Index %u was not valid)",
+                     endpointId, ChipLogValueMEI(clusterId), zeroBasedArrayIndex);
```

Comment:
Suggest making a new BUILD.gn config about "cluster error blind logging" which would be disabled by default, but enabled on some examples. We should not count towards "features" the flash cost of logging that maybe nobody will ever look at. If this log happens in production, you are screwed anyway. Nobody sees the lightbulb's logs :)